### ISSUE TITLE
Match.content() returns "" instead of null

### ISSUE THREAD

**Issue by truelixin** [jOOQ/jOOX#145]
As the document said, 

```java
    /**
     * Get the XML content of the first element in the set of matched elements,
     * or <code>null</code> if there are no matched elements
     * <p>
     * This is the same as calling <code>content(0)</code>
     */
    String content();
``` 

Match.content() should return `null` if no matched elements found, but now it returns `""` (empty string) here.

``` java
private final String content(Element element) {
    if (element == null) {
        return "";
    }
    ....
```

**Comment by lukaseder** [comment]
Thank you very much for your report. Intuitively, I'd say the documentation is wrong as this method should be safely expected to never return null. Will investigate soon.

**Comment by truelixin** [comment]
I suggest return null instead of "", since the content() result is not to be chainable, and with null is easy to check whether the element exists or not.

**Comment by lukaseder** [comment]
Why abuse `Match.content()` for existence checks (a *very* expensive operation in case of existence!) when there is `Match.isEmpty()`?

**Comment by truelixin** [comment]
Since I need the content.
``
$(parent).children().forEach($child -> {
    String content = $child.child("name").content();
    if (content == null)
        ....
    else
        ....
});
``
otherwise
``
$(parent).children().forEach($child -> {
    Match $childName = $child.child("name");
    if ($childName.isEmpty())
        ....
    else {
        String content = $childName.content();
        ....
    }
});
``
I need to store ``$childName`` such a variable.
Am I right? Or better approach?

**Comment by lukaseder** [comment]
OK, I see. I'll have to think about this.

I checked, the `Match.text()` method does return `null` as specified by the Javadoc, so there's certainly an inconsistency here. Making this consistent will break backwards compatibility in any case, so perhaps we could really return `null` from `content()`.

Likewise, I just checked what `jQuery.html()` does, when jQuery doesn't match anything (e.g. `jQuery(".non-matching-selector").html()`. It also returns a JavaScript `null` value.
